layouts/community/ortho_4x12/neo2marin/keymap.c

```diff
@@ -0,0 +1,173 @@
+  //|-------+-------+-------+-------+-------+-------|    |-------+-------+-------+-------+-------+-------|
+      ____  , ____  , ____  , ____  , ____  , ____  ,      ____  , ____  , ____  , ____  , ____  , ____
+  //`-------+-------+-------+-------+-------+-------'    `-------+-------+-------+-------+-------+-------'
+  ),
+
+  [L_MOV] = LAYOUT_ortho_4x12(
+  //,-------+-------+-------+-------+-------+-------.    ,-------+-------+-------+-------+-------+-------.
+      ____  ,KC_PGUP,KC_BSPC, KC_UP ,KC_DEL, KC_PGDN,      ____  , KC_7  , KC_8  , KC_9  ,DE_PLUS,DE_MINS,
+  //|-------+-------+-------+-------+-------+-------|    |-------+-------+-------+-------+-------+-------|
+      ____  ,KC_HOME,KC_LEFT,KC_DOWN,KC_RGHT,KC_END ,      ____  , KC_4  , KC_5  , KC_6  ,DE_COMM, DE_DOT,
+  //|-------+-------+-------+-------+-------+-------|    |-------+-------+-------+-------+-------+-------|
+      ____  ,KC_ESC ,KC_TAB ,KC_INS, KC_ENTER,____  ,      KC_0  , KC_1  , KC_2  , KC_3  ,DE_SCLN, ____  ,
+  //|-------+-------+-------+-------+-------+-------|    |-------+-------+-------+-------+-------+-------|
+      ____  , ____  , ____  , ____  , ____  , ____  ,      ____  , KC_0  , ____  , ____  , ____  , ____ 
+  //`-------+-------+-------+-------+-------+-------'    `-------+-------+-------+-------+-------+-------'
+  ),
+
+  [L_FUN] = LAYOUT_ortho_4x12(
+  //,-------+-------+-------+-------+-------+-------.    ,-------+-------+-------+-------+-------+-------.
+      ____  , AU_ON ,KC_VOLD,ME_MAXM,KC_VOLU, ____  ,     DM_REC1, KC_F7 , KC_F8 , KC_F9 , KC_F12,KC_PSCR,
+  //|-------+-------+-------+-------+-------+-------|    |-------+-------+-------+-------+-------+-------|
+      ____  ,KC_MSTP,KC_MPRV,KC_MPLY,KC_MNXT, NEO2  ,     DM_RSTP, KC_F4 , KC_F5 , KC_F6 , KC_F11,KC_BRK ,
+  //|-------+-------+-------+-------+-------+-------|    |-------+-------+-------+-------+-------+-------|
+      ____  , AU_OFF,KC_TAB ,KC_INS ,KC_ENTER,QWERTZ,     DM_PLY1, KC_F1 , KC_F2 , KC_F3 , KC_F10, ____  ,
+  //|-------+-------+-------+-------+-------+-------|    |-------+-------+-------+-------+-------+-------|
+      ____  , ____  , ____  , ____  , ____  , ____  ,      ____  , ____  , ____  , ____  , ____  , ____
+  //|-------+-------+-------+-------+-------+-------|    |-------+-------+-------+-------+-------+-------|
+  ),
+
+};
+
+#if defined(AUDIO_ENABLE)
+float SONG_QWERTZ[][2] = SONG(QWERTY_SOUND);
+float SONG_NEO2[][2]   = SONG(DVORAK_SOUND);
+#endif
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTZ:
+      if (record->event.pressed) {
+        default_layer_set(L_QWZ);
+        #if defined(AUDIO_ENABLE)
+          PLAY_SONG(SONG_QWERTZ);
+        #endif
+      }
+      return false;
+      
+    case NEO2:
+      if (record->event.pressed) {
+        default_layer_set(L_NEO);
+        #if defined(AUDIO_ENABLE)
+          PLAY_SONG(SONG_NEO2);
+        #endif
+      }
+      return false;
```

Comment:
the `DF` keycode can handle these.  

Also:
https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

Reply:
Unfortunately, I could not get the `DF` keycode to play any song as specified by `#define DEFAULT_LAYER_SONGS`.

To try it, I added these lines to my `config.h`:
```c
#define DEFAULT_LAYER_SONGS \
    { SONG(DVORAK_SOUND), SONG(QWERTY_SOUND) }
```
and used the keycodes `DF(L_NEO)` and `DF(L_QWZ)` instead of my custom keycodes.
Then, the default layer can be correctly switched, but no sound is being played on switch.

So as long as the `DF` keycode does not produce any sound, these custom keycodes with explicit `PLAY_SONG()` calls are obligatory for me.